website/docs/transforms/rename.md

```diff
@@ -17,37 +17,137 @@ Add the following configuration to your Mesh config file:
 ```yml
 transforms:
   - rename:
-      - from: 
-          type: ApiUser
-        to: 
-          type: User
-      - from: 
-          type: Query
-          field: apiUser
-        to: 
-          type: Query
-          field: user
+      mode: bare | wrap
+      renames:
+        - from:
+            type: ApiUser
+          to:
+            type: User
+        - from:
+            type: Query
+            field: apiUser
+          to:
+            type: Query
+            field: user
 ```
 
 or you can use regular expressions to rename multiple types, fields or both
 
 ```yml
   - rename:
+      mode: bare | wrap
+      renames:
+        - from:
+            type: Api(.*)
+          to:
+            type: $1
+          useRegExpForTypes: true
 
-      - from: 
-          type: Api(.*)
-        to: 
-          type: $1
-        useRegExpForTypes: true
+        - from:
+            type: Query
+            field: api(.*)
+          to:
+            type: Query
+            field: $1
+          useRegExpForFields: true
 
-      - from: 
-          type: Query
-          field: api(.*)
-        to: 
-          type: Query
-          field: $1
-        useRegExpForFields: true
+```
+
+## Two different modes
+Although apparently both `bare` and `wrap` modes do achieve the same result, when it comes to renaming; they are very different.  
+Let's take a look at how they operate.
+
+### Wrap
+Wrap applies transformations by adding a wrapping layer to the original GraphQL schema. The handler generates a GraphQL
+ schema and passes it to the rename transform. `wrap` receives this schema and rather than updating it, it will apply a layer on top of it, which scope is to provide the values exposed by the names of the original schema to the new names that you define as `renames`.  
+This approach is safe as we have used it extensively in `graphql-tools`. However, it might have some runtime implications, since when receiving requests and preparing responses, the wrapping layer needs to apply the `renames` on the fly for every GraphQL operation.
+
+> NOTE: `wrap` is the only approach that works when you want to transform at all-sources (root) level, unless you're using [merger-bare](https://graphql-mesh.com/docs/api/modules/merger-bare/). If you want to remove the possible runtime implications, consider either moving your `renames` at the data source level or opt into `merger-bare`; in order to take advantage of `bare` mode.
+
+Example:
+```yaml
+sources:
+  - name: Countries
+    handler:
+      soap:
+        wsdl: http://webservices.../wso?WSDL
+  - name: Users
+    handler:
+      openapi:
+        source: https://api.../swagger.yaml
+    transforms:
+      rename:
+          mode: wrap # you can use wrap here, but consider using "bare" to remove runtime cost
```

Comment:
If you don't want to talk about "cost", consider changing this to "implications" as well; for consistency.